scss/_patterns_navigation.scss

```diff
@@ -25,6 +25,7 @@
       $border-color: $color-mid-light,
       $text-color: $color-dark
     );
+    border-bottom: 1px solid $color-mid-light;
```

Comment:
Yes, all breakpoints - see: https://github.com/ubuntudesign/vanilla-design/blob/master/Navigation/navigation.png

As this is the exception, not the rule, I don't think an amendment to the mixin is necessary.